### ISSUE TITLE
Default avatar isn't able to be saved, doing so deletes head

### ISSUE THREAD

**Issue by TehTurk** [Yellow-Dog-Man/Resonite-Issues#331]
### Describe the bug?

Found out while testing for finding an avatar root. It occurs when you try to save the default avatar when you have no avatar set as a favorite. 

### To Reproduce

- Make sure any avatar isn't equipped or favorited and respawn yourself
- You have entered the default avatar
- Proceed to save avatar
- This won't occur and your head will delete causing issues to new users. 

### Expected behavior

It should save the avatar, or at the very least, have the default avatar prefab be the default avatar. 

### Resonite Version Number

Current

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Index

### Log Files

Not needed

### Additional Context

Written and submitted from mobile

### Reporters

- Turk
- marsmanntje

**Comment by shadowpanther** [comment]
Saving the avatar dequips it from the user for one update.
Dequipping the default head and hands parts cause them to self-destroy.
This is made so that the default avatar doesn't stay behind when the user enters another avatar.

So, everything kinda works as it should. Except when there's no head part, the user should get a new default head spawned.

**Comment by TehTurk** [comment]
Yeah I kind of figured this was what was happening, but this could also be saved if the default one from the essentials was used as a prefab vs the hard baked one. It was always a funny issue people could save the default in a normal way.

**Comment by shiftyscales** [comment]
Related: #3583
This is not technically a bug- as mentioned above- you are technically leaving the avatars while you attempt to save them- so `AvatarDestroyOnDequip` fires and destroys the avatars you have equipped.

I understand why the current behaviour is undesirable however, so this can be looked into at some point.
